src/repr/src/relation.rs

```diff
@@ -569,15 +569,16 @@ impl Arbitrary for RelationDesc {
     type Strategy = BoxedStrategy<RelationDesc>;
 
     fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
-        let num_columns = Union::new_weighted(vec![
-            (100, Just(0..4)),
-            (50, Just(4..8)),
-            (25, Just(8..16)),
-            (12, Just(16..32)),
-            (6, Just(32..64)),
-            (3, Just(64..128)),
-            (1, Just(128..256)),
-        ]);
+        let mut weights = vec![(100, Just(0..4)), (50, Just(4..8)), (25, Just(8..16))];
+        if std::env::var("PROPTEST_LARGE_DATA").is_ok() {
```

Comment:
I'll add some infra in `mz_ore` around `PROPTEST_LARGE_DATA` but for now this helps prevent OOMs for long running cargo-stress